### PR TITLE
fix: image scanning flows

### PR DESCRIPTION
**Pull request by Shivam-nagar23**
# Description

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1541

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/policyGovernance/security/imageScanning/repository/ImageScanHistoryRepository.go [hunk 1]
```diff
@@ -163,3 +164,13 @@ func (impl ImageScanHistoryRepositoryImpl) FindByImage(image string) (*ImageScan
 		Where("image = ?", image).Order("execution_time desc").Limit(1).Select()
 	return &model, err
 }
+
+func (impl ImageScanHistoryRepositoryImpl) FindByImageAndDigestWithState(imageDigest string, image string) (*ImageScanExecutionHistory, error) {
```
Comment (prakash100198) [inline pkg/policyGovernance/security/imageScanning/repository/ImageScanHistoryRepository.go:168]: change func name , impl is not acc to func name

File: pkg/policyGovernance/security/imageScanning/ImageScanService.go [hunk 2]
```diff
@@ -249,13 +257,40 @@ func (impl ImageScanServiceImpl) fetchImageExecutionHistoryMapByIds(historyIds [
...
+	if request.AppId > 0 && request.EnvId > 0 {
+		artifactId, err := impl.fetchResourceScanningDataForEnv(request.AppId, request.EnvId)
+		if err != nil {
+			impl.Logger.Errorw("error while fetching scan execution result", "appId", request.AppId, "envId", request.EnvId, "err", err)
+			return nil, err
+		}
+		// setting latest artifact deploy on app and env
+		request.ArtifactId = artifactId
+	}
 	if request.ImageScanDeployInfoId > 0 {
```
Comment (prakash100198) [inline pkg/policyGovernance/security/imageScanning/ImageScanService.go:294]: add todo, that if request.ImageScanDeployInfoId > 0 check is to be removed after few sprints

File: pkg/policyGovernance/security/imageScanning/ImageScanService.go [hunk 3]
```diff
@@ -249,13 +257,40 @@ func (impl ImageScanServiceImpl) fetchImageExecutionHistoryMapByIds(historyIds [
 	}
 	return mapOfExecutionHistoryIdVsExecutionTime, nil
 }
+func (impl ImageScanServiceImpl) fetchResourceScanningDataForEnv(appId, envId int) (int, error) {
```
Comment (prakash100198) [inline pkg/policyGovernance/security/imageScanning/ImageScanService.go:260]: func name and implementation is different, we can change the func name

**Review by prakash100198 — CHANGES_REQUESTED** [review]

File: pkg/policyGovernance/security/imageScanning/repository/ImageScanHistoryRepository.go (hunk 1 above)
Comment (Shivam-nagar23) [reply pkg/policyGovernance/security/imageScanning/repository/ImageScanHistoryRepository.go:168]: DONE

File: pkg/policyGovernance/security/imageScanning/ImageScanService.go (hunk 2 above)
Comment (Shivam-nagar23) [reply pkg/policyGovernance/security/imageScanning/ImageScanService.go:294]: DONE

File: pkg/policyGovernance/security/imageScanning/ImageScanService.go (hunk 3 above)
Comment (Shivam-nagar23) [reply pkg/policyGovernance/security/imageScanning/ImageScanService.go:260]: DONE

File: api/resourceScan/resourceScanRouter.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
```
Comment (nishant-d) [inline api/resourceScan/resourceScanRouter.go:2]: add licence

File: pkg/policyGovernance/security/imageScanning/adapter/adapter.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	return parsedVulnerabilities
+
+}
+
+func BuildImageVulnerabilityResponse(image string, vulnerabilities parser.Vulnerabilities, metadata parser.Metadata) *parser.ImageVulnerability {
+	return &parser.ImageVulnerability{Image: image, Vulnerabilities: vulnerabilities, Metadata: metadata}
+}
+
+func BuildMetadata(status string, startedOn time.Time, scanToolName string) parser.Metadata {
```
Comment (nishant-d) [inline pkg/policyGovernance/security/imageScanning/adapter/adapter.go:40]: use pointer

File: pkg/policyGovernance/security/imageScanning/helper/parser/types.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+func (r Severity) ToString() string {
+	return string(r)
+}
+
+type Summary struct {
+	Severities map[Severity]int `json:"severities"`
+}
+
+func (summary *Summary) String() string {
+	return fmt.Sprintf("%d Critical, %d High, %d Medium, %d Low, %d Unknown", summary.Severities[CRITICAL], summary.Severities[HIGH], summary.Severities[MEDIUM], summary.Severities[LOW], summary.Severities[UNKNOWN])
```
Comment (nishant-d) [inline pkg/policyGovernance/security/imageScanning/helper/parser/types.go:31]: if not used removed

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/policyGovernance/security/imageScanning/adapter/adapter.go (hunk 5 above)
Comment (Shivam-nagar23) [reply pkg/policyGovernance/security/imageScanning/adapter/adapter.go:40]: DONE

File: pkg/policyGovernance/security/imageScanning/helper/parser/types.go (hunk 6 above)
Comment (Shivam-nagar23) [reply pkg/policyGovernance/security/imageScanning/helper/parser/types.go:31]: DONE

File: api/resourceScan/resourceScanRouter.go (hunk 4 above)
Comment (Shivam-nagar23) [reply api/resourceScan/resourceScanRouter.go:2]: DONE

**Review by nishant-d — APPROVED** [review]

**Pull request merged**